From e001251cb5708c406d477eca6aa912e5692b17fe Mon Sep 17 00:00:00 2001
From: Peter Oh <peter.oh@bowerswilkins.com>
Date: Tue, 17 Apr 2018 21:55:05 -0700
Subject: [PATCH 08/16] mesh: set interface type to mesh before setting
 interface

Correct interface type is required to start DFS CAC that can be
triggered during interface setup.

Signed-off-by: Peter Oh <peter.oh@bowerswilkins.com>
---
 wpa_supplicant/mesh.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/wpa_supplicant/mesh.c
+++ b/wpa_supplicant/mesh.c
@@ -339,14 +339,14 @@ static int wpa_supplicant_mesh_init(stru
 
 	wpa_supplicant_conf_ap_ht(wpa_s, ssid, conf);
 
-	if (hostapd_setup_interface(ifmsh)) {
-		wpa_printf(MSG_ERROR,
-			   "Failed to initialize hostapd interface for mesh");
+	if (wpa_drv_init_mesh(wpa_s)) {
+		wpa_msg(wpa_s, MSG_ERROR, "Failed to init mesh in driver");
 		return -1;
 	}
 
-	if (wpa_drv_init_mesh(wpa_s)) {
-		wpa_msg(wpa_s, MSG_ERROR, "Failed to init mesh in driver");
+	if (hostapd_setup_interface(ifmsh)) {
+		wpa_printf(MSG_ERROR,
+			   "Failed to initialize hostapd interface for mesh");
 		return -1;
 	}
 
